vcf.c

```diff
@@ -2367,9 +2383,15 @@ static int vcf_parse_format(kstring_t *s, const bcf_hdr_t *h, bcf1_t *v, char *p
         while ( t < end )
         {
             fmt_aux_t *z = &fmt[j++];
+            if (!z->buf) {
+                hts_log_error("Memory allocation failure for FORMAT field type %d",
+                              z->y>>4&0xf);
+                v->errcode |= BCF_ERR_LIMITS;
+                return -1;
+            }
             if ((z->y>>4&0xf) == BCF_HT_STR) {
                 if (z->is_gt) { // genotypes
-                    int32_t is_phased = 0, *x = (int32_t*)(z->buf + z->size * m);
+                    int32_t is_phased = 0, *x = (int32_t*)(z->buf + z->size * (size_t)m);
```

Comment:
`m` is a local variable. Rather than all this casting, can it be declared more accurately as `size_t m` instead? This might trickle down to `fmt_aux_t::max_m` and/or `fmt_aux_t::size` and suggest changing them to `size_t` too, but that doesn't seem like a bad thing…

Reply:
The function is so vast I lose track of the scope of these variables, so I just went with the minimalist change.

I'll take a look at changing the type for local var m.  I don't wish to start changing data types though as that requires a deeper understanding of vcf which I don't currently have time to, nor the desire due to the cost of -10 sanity points!

Reply:
Hmm looking at it `m` is overloaded.   The bit of code you're referring to uses m as sample id.  In other places it's just listed as a vector (unspecified which).  If I change type of m, should I also change type of g? of r? of l? or j? etc.  I could create a new var `sid` say for sample id and use that throughout the specific block of code you refer to, but it's just as much code changing to read.

I think I'll keep this as is and request that @pd3 looks at changing types if he feels it is valid.  I simply don't understand all the knock on implications of changing type here given the overloading of var names.

Reply:
Aarrgggghhhhh… fair enough